### DISCUSSION TITLE
Formally decide if Cyber/Mountain Duck will support Requester Pays S3 buckets

### DISCUSSION THREAD

**Discussion by ftrotter** [iterate-ch/cyberduck#14470] [Q&A]
I would like to access an S3 bucket that is configured for "requester pays". [Here is the AWS Documentation for the feature](https://docs.aws.amazon.com/AmazonS3/latest/userguide/RequesterPaysBuckets.html), which allows for one S3 user to pay for uploading and storing data, but another user to pay for downloading the data. 

Not only can I not figure out how to do this with the Mountain Duck configuration option, when I see the history of the feature on this project, I find a very ambiguous ticket history indicating that it "is", "is not" but "maybe is supported". #13727  #9807 #8893  

It is totally reasonable to decide that a given feature is not something that a project is going to support. But you have tags called "wontfix" and "lowpriority" for a reason. It helps people like me, who are trying to understand if this is a feature that we can look forward to, know if we need to look elsewhere for a project that is going to support this feature. 

It also allows outside developers to see that there is a potential feature that they could contribute to your codebase that either would, or would not be welcome.. 

But as it is, I cannot easily tell where this feature stands. Is this on the roadmap? Is it welcome? Is it something that will never be worked on? Is there a reason why? 

This is the fourth time that your users have requested clarity regarding a feature, please consider responding in a manner that will allow for future users to know where they stand on this, rather than having to create a new issues to request the same information again. 

Thanks for the work that you do on this project. I use your tools frequently, and find that they generally work well. 

Regards,
-FT

**Comment by dkocher** [comment]
#13727 refers to Google Storage which has the same feature which is enabled depending on the bucket configuration.

**Comment by dkocher** [comment]
#9807 is a duplicate confirming that the option is supported again as reported in #8893 as of version [4.7.1](https://github.com/iterate-ch/cyberduck/milestone/76). This is configurable with the [option](https://docs.cyberduck.io/cyberduck/preferences/#hidden-configuration-options) `s3.bucket.requesterpays` enabled by default [^1].

[^1]: https://github.com/iterate-ch/cyberduck/blob/eeb797fa55c7661638d57a8f8fab9d0f39d9b9a0/defaults/src/main/resources/default.properties#LL287-L287C14

**Comment by dkocher** [comment]
I have opened a documentation issue [^1].

[^1]: iterate-ch/docs#362

**Comment by ftrotter** [comment]
Thanks for the progress on this!! Am I correct that it is something that is configurable.. but not something that is part of the GUI?

If so, then would it be appropriate to make a feature ticket for GUI support too? 

Assuming it does not actually require anything but GUI development, that might be simple enough that I might be able to wrangle up a pull request...

**Comment by Cassie205** [comment]
Yes, that's right. You can open a feature ticket for the GUI option, but this currently doesn't have a high priority for now on our side. Contributions through pull requests are always welcome but need some upfront discussions as to where the changes should be added within the code.

The option is enabled by default but can be manually disabled with the help of the hidden property.
